cli/azd/pkg/auth/manager_test.go

```diff
@@ -246,25 +254,76 @@ func TestLoginDeviceCode(t *testing.T) {
 	require.True(t, errors.Is(err, ErrNoCurrentUser))
 }
 
-func newMemoryConfigManager() config.UserConfigManager {
-	return &memoryConfigManager{
+func TestAuthFileConfigUpgrade(t *testing.T) {
+	cfgMgr := newMemoryConfigManager()
+	userCfg := config.NewConfig(nil)
+	userCfgMgr := newMemoryUserConfigManager()
+
+	err := userCfg.Set(cCurrentUserKey, &userProperties{
+		HomeAccountID: convert.RefOf("homeAccountID"),
+	})
+	require.NoError(t, err)
+
+	err = userCfgMgr.Save(userCfg)
+	require.NoError(t, err)
+
+	m := &Manager{
+		configManager:     cfgMgr,
+		userConfigManager: userCfgMgr,
+		publicClient:      &mockPublicClient{},
+	}
+
+	cfg, err := m.readAuthConfig()
+	require.NoError(t, err)
+
+	properties, err := readUserProperties(cfg)
+	require.NoError(t, err)
+	require.NotNil(t, properties.HomeAccountID)
+	require.Equal(t, "homeAccountID", *properties.HomeAccountID)
```

Comment:
nit here: Can we also validate `userCfg` no longer contains the properties? (Or maybe that's not necessary)